custom_components/opnsense/coordinator.py

```
@@ -70,162 +92,180 @@
-            # copy the old data to have around
-            current_time: float = time.time()
-
-            previous_state: MutableMapping[str, Any] = copy.deepcopy(self._state)
-            if "previous_state" in previous_state:
-                del previous_state["previous_state"]
-
-            # ensure clean state each interval
-            self._state = {}
-            self._state["update_time"] = current_time
-            self._state["previous_state"] = previous_state
+    def _build_categories(self) -> list[MutableMapping[str, str]]:
```

Pylint (3.3.7)

[refactor] Too many branches (13/12)

(R0912)